src/util.rs

```diff
@@ -327,3 +327,16 @@ pub fn split_repo_aur_pkgs<S: AsRef<str> + Clone>(config: &Config, pkgs: &[S]) -
 
     (repo, aur)
 }
+
+pub fn repo_aur_pkgs(config: &Config) -> (Vec<alpm::Package<'_>>, Vec<alpm::Package<'_>>) {
+    if config.repos != LocalRepos::None {
+        let (repo, aur) = repo::repo_aur_dbs(config);
+        let repo = repo.iter().flat_map(|db| db.pkgs()).collect::<Vec<_>>();
+        let aur = aur.iter().flat_map(|db| db.pkgs()).collect::<Vec<_>>();
+        (repo, aur)
+    } else {
+        let mut pkgs = config.alpm.localdb().pkgs().iter().collect::<Vec<_>>();
+        pkgs.retain(|pkg| config.alpm.syncdbs().pkg(pkg.name()).is_err());
+        (Vec::new(), pkgs)
```

Comment:
You need to fill in the repo packages here. In this case aur is the inverse of repo. So you can just do `config.alpm.localdb().pkgs().iter()` then filter with `config.alpm.syncdbs().pkg(pkg.name()).is_err()` for aur packages and `config.alpm.syncdbs().pkg(pkg.name()).is_ok()` for repo packages.